iroh-gateway/src/main.rs

```diff
@@ -1,40 +1,78 @@
+use std::path::PathBuf;
+
 use anyhow::Result;
 use clap::Parser;
 use iroh_gateway::{
     config::{Config, RpcConfig},
     core::Core,
     metrics,
 };
+use iroh_util::{from_toml_file, iroh_home_path};
 use prometheus_client::registry::Registry;
 
+const CONFIG: &str = "gateway.config.toml";
+
 #[derive(Parser, Debug, Clone)]
 #[clap(author, version, about, long_about = None)]
 struct Args {
-    #[clap(short, long, required = false, default_value_t = 9050)]
-    port: u16,
     #[clap(short, long)]
-    writeable: bool,
+    port: Option<u16>,
+    #[clap(short, long)]
+    writeable: Option<bool>,
     #[clap(short, long)]
-    fetch: bool,
+    fetch: Option<bool>,
     #[clap(short, long)]
-    cache: bool,
+    cache: Option<bool>,
     #[clap(long = "no-metrics")]
     no_metrics: bool,
+    #[clap(long)]
+    cfg: Option<PathBuf>,
+}
+
+impl Args {
+    fn override_config(&self, mut cfg: Config) -> Config {
```

Comment:
you shouldn't need to implement this manually if you use https://docs.rs/config with different sources